source/common/redis/proxy_filter.cc

```diff
@@ -19,9 +19,15 @@ ProxyFilterConfig::ProxyFilterConfig(const Json::Object& config, Upstream::Clust
       cluster_name_(config.getString("cluster_name")),
       stat_prefix_(fmt::format("redis.{}.", config.getString("stat_prefix"))),
       stats_(generateStats(stat_prefix_, scope)) {
-  if (!cm.get(cluster_name_)) {
-    throw EnvoyException(
-        fmt::format("redis filter config: unknown cluster name '{}'", cluster_name_));
+  Upstream::ThreadLocalCluster* cluster = cm.get(cluster_name_);
+  if (cluster == nullptr) {
+    throw EnvoyException(fmt::format("redis filter config: unknown cluster '{}'", cluster_name_));
+  }
+
+  if (cluster->info()->addedViaApi()) {
```

Comment:
RDS should "work" in that it won't crash (I think), but I think people would generally find this behavior extremely confusing if they accidentally deleted the RDS cluster.

Anyhow, I guess I was erring on the side of trying to raise issues to users faster to avoid support burden (and potentially allow people to build things in a more simplistic way potentially if they want), but given that redis is not officially supported yet, I can revert this part of the change if you don't think it's useful in general. I don't feel that strongly about it. Let me know. Note that if we revert this part of the change I will definitely work on fixing the redis code to support cluster removal since it absolutely will crash in bad ways right now.